packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -255,18 +257,47 @@ fn process_fn_items(
         None => quote! {},
     };
 
-    let contract = match &manifest.contract_id {
-        Some(contract_id) => {
+    let contracts = match &manifest.contract_id {
+        ContractIds::Single(contract_id) => match contract_id {
+            Some(contract_id) => {
+                quote! {
+                    let manifest_contract_id = Bech32ContractId::from_str(#contract_id).expect("Failed to parse manifest 'contract_id' as Bech32ContractId");
+                    let receipt_contract_id = Bech32ContractId::from(id);
+                    if receipt_contract_id != manifest_contract_id {
+                        Logger::info("Not subscribed to this contract. Will skip this receipt event. <('-'<)");
+                        continue;
+                    }
+                }
+            }
+            None => quote! {},
+        },
+        ContractIds::Multiple(contract_ids) => {
+            let contract_id_checks = contract_ids.iter().map(|contract_id| {
+            match contract_id {
+                Some(contract_id) => {
+                    quote! {
+                        let manifest_contract_id = Bech32ContractId::from_str(#contract_id).expect("Failed to parse manifest 'contract_id' as Bech32ContractId");
+                        if receipt_contract_id == manifest_contract_id {
+                            found = true;
+                        }
+                    }
+                }
+                None => quote! {},
+            }
+        });
```

Comment:
It would be more efficient to instantiate a `HashSet` from the contract IDs and check whether it contains the receipt contract ID.